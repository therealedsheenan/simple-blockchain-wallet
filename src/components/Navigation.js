import React, { Component } from "react";
import { Layout, Button } from "zent";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";

import { isAuthenticated } from "../utils";
import { postUnauthRequest } from "../modules/auth/actions";

const { Row, Col } = Layout;

// navigation menu
export class Navigation extends Component {
  static propTypes = {
    postUnauthRequest: PropTypes.func.isRequired // eslint-disable-line
  };

  state = {
    user: {}
  };

  render() {
    const { state, props } = this;
    return (
      <div className="Navigation theme-primary-2">
        <Row>
          <Col span={12}>
            <Link to={`${isAuthenticated() ? "/wallets" : "/"}`}>
              <h1>Simple blockchain wallet</h1>
            </Link>
          </Col>
          <Col span={12}>
            {isAuthenticated() && (
              <div className="Navigation-user">
                <Link to="/wallets/send">Send Bitcoin</Link>
                <Link to="/wallets">Wallets</Link>
                <span>{state.user.username}</span>
                <Button type="danger" outline onClick={props.postUnauthRequest}>
                  Logout
                </Button>
              </div>
            )}
          </Col>
        </Row>
      </div>
    );
  }
}

export default connect(
  ({ auth }) => ({ auth }),
  dispatch => ({
    postUnauthRequest: () => dispatch(postUnauthRequest())
  })
)(Navigation);
