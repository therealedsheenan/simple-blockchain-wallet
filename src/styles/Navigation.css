.Navigation {
  padding: 16px 20px;
  color: #f8f8f8;
  background-color: #444;
}

.Navigation h1 {
  color: #f8f8f8;
}

.Navigation-user {
  text-align: right;
}

.Navigation-user a {
  color: #f8f8f8;
  margin: 0 10px;
}

.Navigation-user button {
  margin-left: 10px;
}
