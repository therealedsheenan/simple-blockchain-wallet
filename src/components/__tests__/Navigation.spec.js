import React from "react";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import { Switch, Route, BrowserRouter } from "react-router-dom";

import { Navigation } from "../Navigation";
import mockLocalStorage from "../../__mocks__/localStorage";

describe("Login component", () => {
  beforeEach(() => {
    mockLocalStorage();
  });

  const mockPostUnauthRequest = jest.fn();
  const NavigationMock = (
    <BrowserRouter>
      <Switch>
        <Route>
          <Navigation postUnauthRequest={mockPostUnauthRequest} />
        </Route>
      </Switch>
    </BrowserRouter>
  );

  it("Should match snapshot", () => {
    const component = mount(NavigationMock);

    expect(toJson(component)).toMatchSnapshot();
  });
  // username and password
  // it("Should contain username and password input", () => {
  //   const component = mount(<NavigationMock />);
  //   expect(component.text()).toContain("Username");
  //   expect(component.text()).toContain("Password");
  // });
});
