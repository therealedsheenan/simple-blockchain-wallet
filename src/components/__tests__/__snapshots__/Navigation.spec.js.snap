// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component Should match snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Switch>
      <Route
        computedMatch={
          Object {
            "isExact": true,
            "params": Object {},
            "path": "/",
            "url": "/",
          }
        }
        location={
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          }
        }
      >
        <Navigation
          postUnauthRequest={[Function]}
        >
          <div
            className="Navigation theme-primary-2"
          >
            <Row
              prefix="zent"
            >
              <div
                className="zent-row"
              >
                <Col
                  prefix="zent"
                  span={12}
                >
                  <div
                    className="zent-col zent-col-12"
                  >
                    <Link
                      replace={false}
                      to="/"
                    >
                      <a
                        href="/"
                        onClick={[Function]}
                      >
                        <h1>
                          Simple blockchain wallet
                        </h1>
                      </a>
                    </Link>
                  </div>
                </Col>
                <Col
                  prefix="zent"
                  span={12}
                >
                  <div
                    className="zent-col zent-col-12"
                  />
                </Col>
              </div>
            </Row>
          </div>
        </Navigation>
      </Route>
    </Switch>
  </Router>
</BrowserRouter>
`;
